format/Schema.fbs

```diff
@@ -44,9 +44,15 @@ table Null {
 table Struct_ {
 }
 
+/// List with 32-bit offsets
 table List {
 }
 
+// List with 64-bit offsets. This type is optional and currently
+// only supported by the C++ implementation.
+table LargeList {
```

Comment:
@fsaintjacques had a suggestion which I liked and I think is worth discussing to parameterize List instead of creating a new type.  I know @wesm suggested he prefers this as a separate type like you have here, but I'm not sure what the rationale is.